ariefrahmansyah/oop_encapsulation.py

```diff
@@ -0,0 +1,24 @@
+class Computer:
+
+    def __init__(self):
+        self.__maxprice = 900
+
+    def sell(self):
+        print("Selling Price: {}".format(self.__maxprice))
+
+    def setMaxPrice(self, price):
+        self.__maxprice = price
+
+
+c = Computer()
+c.sell()
+
+# change the max price by changing the attribute directly
+# won't work!
+c.__maxprice = 1000
+c.sell()
```

Comment:
if I add `print(c.__maxprice)` after `c.__maxprice = 1000`, it will print 1000. Why is that?

Reply:
Can you share your full code?

Reply:
Here:

```python
class Computer:

    def __init__(self):
        self.__maxprice = 900

    def sell(self):
        print("Selling Price: {}".format(self.__maxprice))

    def setMaxPrice(self, price):
        self.__maxprice = price


c = Computer()
c.sell()

# change the max price by changing the attribute directly
# won't work!
c.__maxprice = 1000
print("Sellint Price: {}".format(c.__maxprice))
c.sell()

# using setter function
c.setMaxPrice(1000)
c.sell()
```